libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1523,10 +1547,19 @@ BaseType_t xReturn;
 				{
 					strcpy( xDNSCache[ xFreeEntry ].pcName, pcName );
 
-					xDNSCache[ xFreeEntry ].ulIPAddress = *pulIP;
+					xDNSCache[ xFreeEntry ].ulIPAddress[0] = *pulIP;
 					xDNSCache[ xFreeEntry ].ulTTL = ulTTL;
 					xDNSCache[ xFreeEntry ].ulTimeWhenAddedInSeconds = ulCurrentTimeSeconds;
-
+#if( ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY > 1 )
+					xDNSCache[ xFreeEntry ].ucNumIPAddresses = 1;
+					xDNSCache[ xFreeEntry ].ucCurrentIPAddress = 0;
+
+                    /* Initialize all remaining IP addresses in this entry to 0 */
+                    memset( &xDNSCache[ xFreeEntry ].ulIPAddress[ 1 ],
+                            0,
+                            sizeof( xDNSCache[ xFreeEntry ].ulIPAddress[ 1 ] ) *
+                            ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY-1 );
```

Comment:
~~~c
 memset( &xDNSCache[ xFreeEntry ].ulIPAddress[ 1 ],
     0,
     sizeof( xDNSCache[ xFreeEntry ].ulIPAddress[ 1 ] ) *
-    ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY-1 );
+        ( ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY-1 ) );
~~~
@gkwicker, could it be that this expression needed extra brackets?
And an indentation because it is part of the third parameter?

Reply:
Yes, good point – will change.